File: ProjectTest/OneLang/CSharp/One/Transforms/ConvertToMethodCall.cs
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using One.Ast;
using One;

namespace One.Transforms
{
    public class ConvertToMethodCall : AstTransformer {
        public ConvertToMethodCall(): base("ConvertToMethodCall")
        {
            
        }
        
        protected override Expression visitExpression(Expression expr) {
            var origExpr = expr;
            
            expr = base.visitExpression(expr) ?? expr;
            
            if (expr is BinaryExpression && ((BinaryExpression)expr).operator_ == "in")
                expr = new UnresolvedCallExpression(new PropertyAccessExpression(((BinaryExpression)expr).right, "hasKey"), new Type_[0], new Expression[] { ((BinaryExpression)expr).left });
            
            expr.parentNode = origExpr.parentNode;
            return expr;
        }
    }
}

File: ProjectTest/OneLang/Java/src/main/java/OneLang/One/Transforms/ConvertToMethodCall/ConvertToMethodCall.java
package OneLang.One.Transforms.ConvertToMethodCall;

import OneLang.One.Ast.Expressions.Expression;
import OneLang.One.Ast.Expressions.ElementAccessExpression;
import OneLang.One.Ast.Expressions.UnresolvedCallExpression;
import OneLang.One.Ast.Expressions.PropertyAccessExpression;
import OneLang.One.Ast.Expressions.BinaryExpression;
import OneLang.One.Ast.Expressions.StringLiteral;
import OneLang.One.AstTransformer.AstTransformer;

import OneLang.One.AstTransformer.AstTransformer;
import OneLang.One.Ast.Expressions.Expression;
import OneLang.One.Ast.Expressions.BinaryExpression;
import OneStd.Objects;
import OneLang.One.Ast.Expressions.UnresolvedCallExpression;
import OneLang.One.Ast.Expressions.PropertyAccessExpression;
import OneLang.One.Ast.Interfaces.IType;

public class ConvertToMethodCall extends AstTransformer {
    public ConvertToMethodCall()
    {
        super("ConvertToMethodCall");
        
    }
    
    protected Expression visitExpression(Expression expr) {
        var origExpr = expr;
        var visitExpressionResult = super.visitExpression(expr);
        
        expr = visitExpressionResult != null ? visitExpressionResult : expr;
        
        if (expr instanceof BinaryExpression && Objects.equals(((BinaryExpression)expr).operator, "in"))
            expr = new UnresolvedCallExpression(new PropertyAccessExpression(((BinaryExpression)expr).right, "hasKey"), new IType[0], new Expression[] { ((BinaryExpression)expr).left });
        
        expr.parentNode = origExpr.parentNode;
        return expr;
    }
}